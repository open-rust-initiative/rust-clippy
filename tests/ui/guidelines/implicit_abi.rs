// run-rustfix
#![allow(unused)]
#![warn(clippy::implicit_abi)]

use std::ffi::{c_char, c_void};

#[repr(C)]
pub struct External(i8);

#[rustfmt::skip]
extern {
    fn foo();
    fn bar(a: u8, b: i32) -> u64;
    pub fn baz(a: c_char) -> c_void;
    pub fn qux(a: External) -> c_void;

    pub static a: std::ffi::c_int;
    pub static b: External;
    static c: i8;
}

#[rustfmt::skip]
extern
{
    static SOME_PTR: *mut c_void;
}

#[rustfmt::skip]
extern { fn my_c_fn(a: i8) -> c_void; }

// For reference, don't lint
extern "C" {
    fn c_func();
}
// For reference, don't lint
extern "system" {
    pub fn system_call();
}

fn main() {}
