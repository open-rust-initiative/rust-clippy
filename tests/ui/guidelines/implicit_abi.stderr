error: missing ABI label on extern block
  --> $DIR/implicit_abi.rs:11:1
   |
LL | extern {
   | ^^^^^^ help: explicitly states ABI instead: `extern "C"`
   |
   = note: `-D clippy::implicit-abi` implied by `-D warnings`

error: missing ABI label on extern block
  --> $DIR/implicit_abi.rs:23:1
   |
LL | extern
   | ^^^^^^ help: explicitly states ABI instead: `extern "C"`

error: missing ABI label on extern block
  --> $DIR/implicit_abi.rs:29:1
   |
LL | extern { fn my_c_fn(a: i8) -> c_void; }
   | ^^^^^^ help: explicitly states ABI instead: `extern "C"`

error: aborting due to 3 previous errors

